feat(CCDB): add getForRunMiddleTimeStamp method

mpuccio opened the pull request:
Adds a new method `getForRunMiddleTimeStamp` to the `CCDBManagerInstance` class. This method retrieves an object of type `T` from the CCDB using the path and the middle timestamp of the given run number. This is a useful utility method for cases where the exact timestamp does not matter and the run number is available. Especially in O2Physics this could be used in many instances where objects are valid for full runs, and currently the timestamp corresponding to the first BC is used, or sometimes the EOR, causing issues in the analysis.

ddobrigk commented:
thanks a lot @mpuccio ! I think maybe I would even call the function simply `getForRun` instead - the current getter is a bit long in terms of name. :-D 

Once merged we should advertise this further to people who have run-by-run calib or condition objects.

shahor02 commented on CCDB/include/CCDB/BasicCCDBManager.h:
```diff
@@ -311,6 +315,19 @@ T* CCDBManagerInstance::getForTimeStamp(std::string const& path, long timestamp)
+    }
+    return nullptr;
+  }
+  return getForTimeStamp<T>(path, start / 2 + stop / 2);
```
I would suggest to add an optional argument `bool runDependent = false` and return
```
return runDependent ? getForTimeStamp<T>(path, start / 2 + stop / 2) : getSpecific(path, start / 2 + stop / 2, {{"runNumber", std::to_string(runNumber)}});
```

Otherwise, the extraction of e.g. GRPECS object will be ambiguous.

mpuccio replied:
Very good catch, I fixed it!

mpuccio commented:
@ddobrigk good point, I changed the name of the function, @shahor02 I fixed also your comment, thanks a lot to both of you!

shahor02 commented on CCDB/include/CCDB/BasicCCDBManager.h:
```diff
@@ -311,6 +315,19 @@ T* CCDBManagerInstance::getForTimeStamp(std::string const& path, long timestamp)
+    }
+    return nullptr;
+  }
+  return setRunMetadata ? getForTimeStamp<T>(path, start / 2 + stop / 2, {{"runNumber", std::to_string(runNumber)}}) : getForTimeStamp<T>(path, start / 2 + stop / 2);
```
Thanks for fixing my swapped `if` condition. But `getForTimeStamp` does not accept metadata, one should use `getSpecific` instead.

mpuccio replied:
Ah, you are right, sorry! Looking inside `getSpecific` in the end we just need to set the `mMetaData` variable, have a look if it is correct! Thanks a lot!

mpuccio committed:
feat(CCDB): add getForRunMiddleTimeStamp method

Adds a new method `getForRunMiddleTimeStamp` to the `CCDBManagerInstance`
class. This method retrieves an object of type `T` from the CCDB using the
path and the middle timestamp of the given run number. This is a useful
utility method for cases where the exact timestamp does not matter and the
run number is available.
Optionally use metadata in the query
Rename the function
Fix setting of the metadata
